src/lib/isIP.js

```diff
@@ -15,7 +43,28 @@ export default function isIP(str, version = '') {
     const parts = str.split('.').sort((a, b) => a - b);
     return parts[3] <= 255;
   } else if (version === '6') {
-    const blocks = str.split(':');
+    let addressAndZone = [str];
+    // ipv6 addresses could have scoped architecture
+    // according to https://tools.ietf.org/html/rfc4007#section-11
+    if (str.includes('%')) {
+      addressAndZone = str.split('%');
+      if (addressAndZone.length !== 2) {
+        // it must be just two parts
+        return false;
+      }
+      if (!addressAndZone[0].includes(':')) {
+        // the first part must be the address
+        return false;
+      }
+
+      if (addressAndZone[1] === '') {
+        // the second part must not be empty
+        return false;
+      }
+    }
+
+    const address = addressAndZone[0];
+    const blocks = address.split(':');
```

Comment:
`const blocks = addressAndZone[0].split(':');`

Reply:
Just fixed it! :)